src/messaging/ReliableMessageMgr.h

```diff
@@ -102,6 +102,16 @@ class ReliableMessageMgr
      */
     CHIP_ERROR AddToRetransTable(ReliableMessageContext * rc, RetransTableEntry ** rEntry);
 
+    /**
+     *  Calculate the backoff timer for the retransmission.
+     *
+     *  @param[in]   backoffBase    The base interval to use for the backoff calculation, either the active or idle interval.
+     *  @param[in]   sendCount      Count of how many times this message has been sent, including the current attempt.
```

Comment:
If it's "including the current attempt", this counter is always >= 1, so why are we checking for 0 above?

But also, in the spec this seems to be set to 0 for the first transmission as far as I can tell, not 1?  Hard to tell given the spec's phrasing.

And in our impl the way this will work as far as I can tell is that `entry->sendCount` will be 0 after we send the original message and before we send the first retransmit, and then increase after we send each retransmit.

But also, right now this is only being called when computing the delay between original transmission and first retransmit, because that's what `StartRetransmision` does.  `ReliableMessageMgr::ExecuteActions` is where the inter-retransmit delays are computed.

So these comments don't match the implementation (and don't match the spec, I think), and the implementation does not match the spec.

Reply:
Yes, the spec starts with `0`. I'll revise this comment to make clear that count starts with `0` and computes the delay between the last sent message and the next retransmission.

The check for `0` above is because the backoff is linear for the first MRP_BACKOFF_THRESHOLD attempts (2 given start-from-zero counting).

It sounds like the implementation is close to what we want.  We want:

- 1st transmission to be sent without delay, 
- 2nd after backoff(0), 
- 3rd after backoff(1) [note this will be linear equal to backoff(1) due to threshold above], 
- 4th after backoff(2) [exponential increase will start on this one], and so on.

We certainly want spec, implementation, and comments to all match, so thanks very much for pointing out discrepancies.

Reply:
> It sounds like the implementation is close to what we want. We want:

Yes, the impl is doing that, modulo the issue around `ExecuteActions`.

Reply:
Updated the comment.

Reply:
Are changes to `ExecuteActions` planned in this PR, or are we considering that separate work?

Reply:
Added backoff calculation to ExecuteActions.  Sorry for the confusion.